File: front-end/src/app/service-models/app/v1/client/client-employee-attendance-s-m.ts
import { CoinManagementServiceModelBase } from '../../base/coin-management-service-model-base';
import { AttendanceStatusSM } from '../../enums/attendance-status-s-m.enum';

export class ClientEmployeeAttendanceSM extends CoinManagementServiceModelBase<number> {
    attendanceDate!: Date;
    checkIn!: string;
    checkOut!: string;
    attendanceStatus!: AttendanceStatusSM;
    location!: string;
    clientUserId!: number;
    clientCompanyDetailId!: number;
    errorMessageInUpload!: string;
}

File: backend/payroll-backend/Components/Siffrum.Web.Payroll.ServiceModels/v1/Client/ClientEmployeeAttendanceSM.cs
using Siffrum.Web.Payroll.ServiceModels.Base;
using Siffrum.Web.Payroll.ServiceModels.Enums;

namespace Siffrum.Web.Payroll.ServiceModels.v1.Client
{
    public class ClientEmployeeAttendanceSM : SiffrumPayrollServiceModelBase<int>
    {
        public DateTime AttendanceDate { get; set; }
        public string? CheckIn { get; set; }
        public string? CheckOut { get; set; }
        public AttendanceStatusSM AttendanceStatus { get; set; }
        public string Location { get; set; }
        public int ClientUserId { get; set; }
        public int ClientCompanyDetailId { get; set; }
        public string ErrorMessageInUpload { get; set; }

    }
}
